packages/guides/integrating-the-safe-core-sdk.md

```diff
@@ -0,0 +1,353 @@
+
+  This method can receive an array of `MetaTransactionData` objects that represent the multiple transactions we want to include in our MultiSend transaction. If we want to specify some of the optional properties in our MultiSend transaction we can pass a second argument to the method `createTransaction` with the `SafeTransactionOptionalProps` object.
+
+  ```js
+  import { SafeTransactionOptionalProps } from '@gnosis.pm/safe-core-sdk'
+  import { MetaTransactionData } from '@gnosis.pm/safe-core-sdk-types'
+
+  const transactions: MetaTransactionData[] = [
+    {
+      to,
+      data,
+      value,
+      operation
+    },
+    {
+      to,
+      data,
+      value,
+      operation
+    },
+    // ...
+  ]
+
+  const options: SafeTransactionOptionalProps = {
+    safeTxGas, // Optional
+    baseGas, // Optional
+    gasPrice, // Optional
+    gasToken, // Optional
+    refundReceiver, // Optional
+    nonce // Optional
+  }
+
+  const safeTransaction = await safeSdk.createTransaction(transactions, options)
+  ```
+
+
+We can specify the `nonce` of our Safe transaction as long as it is not lower than the current Safe nonce. If multiple transactions are created but not executed they will share the same `nonce` if no `nonce` is specified, making valid the first transaction that gets executed and invalidating all the rest. We can prevent this by calling the method `getNextNonce` from the Safe Service Client instance. This method will take into account all the transactions that are queued and pending to be executed and calculate the next nonce, making it different for all the transactions.
+
+```js
+const nonce = await safeService.getNextNonce(safeAddress)
+```
+
+## 5. Propose the transaction to the service <a name="propose-transaction"></a>
+
+Once we have the Safe transaction object we need to share it with the other owners of the Safe so they can sign it. To send the transaction to the Safe Transaction Service we need to call the method `proposeTransaction` from the Safe Service Client instance and pass an object with the properties:
+- `safeAddress`: The Safe address.
+- `safeTransaction`: The Safe transaction object returned from the method `createTransaction`. Make sure that this object includes the signature of the owner who is proposing it.
+- `safeTxHash`: The Safe transaction hash, calculated by calling the method `getTransactionHash` from the Safe Core SDK.
+- `senderAddress`: The Safe owner proposing the transaction.
+
+```js
+await safeSdk.signTransaction(safeTransaction)
+const safeTxHash = await safeSdk.getTransactionHash(safeTransaction)
+await safeService.proposeTransaction({
+  safeAddress,
+  safeTransaction,
+  safeTxHash,
+  senderAddress
+})
+```
+
+## 6. Get the transaction from the service <a name="get-transaction"></a>
+
+Now the Safe transaction is available on the Safe Transaction Service and the owners need to retrieve it by finding it in the list of pending transactions or getting if by its Safe transaction hash.
```

Comment:
"The transaction is then available on the Safe Transaction Service and the owners can retrieve it by finding it in the pending transaction list, or by getting its Safe transaction hash."